### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by kuu2**
Finished progress bar feature.

Update: cleaned up various things from feedback, removed debugging text, cleaned up cucumber. My cucumber was working a day ago, but for some reason I'm geting the "Can't find #education css" after I changed a few things. Can anyone help?
# Known bugs:
- When you go from finishing the last topic to survey summary, then go back, it will show 120%. I fixed this by making progress be 100% if it's over 100%, but this might have issues if the user decides to add or remove topics.
- When the user adds/removes topics, the progress bar will not update until the new topic questions page is reached.
- Cucumber tests need to be slightly reworked

### CODE REVIEW

File: app/assets/javascripts/startpage/progressBar.coffee [hunk 1]
```diff
@@ -0,0 +1,52 @@
+progressBar = angular.module("ProgressBar", ["StartPageServices"])
+
+progressBar.config(["$routeProvider", ($routeProvider) ->
```
Comment (whsieh) [inline app/assets/javascripts/startpage/progressBar.coffee:3]: Make sure to remove this in your final PR

File: app/assets/javascripts/startpage/progressBar.coffee [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    currentID = StartPageData.getCurrentTopic()
+    percentComplete = 0
+
+    $scope.numQuestions = () -> StartPageData.getNumQuestions()
+    $scope.questionsLeft = () -> StartPageData.getQuestionsLeft()
+    $scope.numTopics = () -> StartPageData.getNumTopics()
+    $scope.isTopicDone = () -> StartPageData.isTopicQuestionsDone(currentID)
+
+    $scope.testvar = () -> return "50%"
```
Comment (whsieh) [inline app/assets/javascripts/startpage/progressBar.coffee]: Make sure to remove this too

File: app/assets/javascripts/startpage/progressBar.coffee [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            percentComplete = 100
+
+        return percentComplete
+
+    $scope.percentCompleteString = () -> # for ng-style
+        n = percentComplete.toString()
+        return n + "%"
+
+    # Updates the # of complete topics variable
+    $scope.updateCompleteTopics = () ->
```
Comment (whsieh) [inline app/assets/javascripts/startpage/progressBar.coffee]: You don't need to bind this to $scope. I would make it local to the controller scope (i.e. just do updateCompleteTopics = () -> ......)

File: features/step_definitions/progress_bar_steps.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+When /I (.*) a response/ do |status|
+    pending "unimplemented"
+end
+
+Then /the progress bar should be at (.*)/ do |progress|
+    page.has_xpath?("//div[@style='width:#{progress}' and @id='progress-bar-percent']")
+end
+
+# Temporary steps
```
Comment (impguard) [inline features/step_definitions/progress_bar_steps.rb]: what's temporary mean?

File: app/assets/javascripts/startpage/progressBar.coffee (hunk 1 above)
Comment (kuu2) [inline app/assets/javascripts/startpage/progressBar.coffee:3]: I'm not sure what you want me to remove here. Can you clarify?

**Pull request closed without merging**